src/components/molecules/CheckboxGroupField/CheckboxGroupField.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import CheckboxField from '../CheckboxField/CheckboxField';
+import Fieldset from '../../atoms/Fieldset/Fieldset';
+import Legend from '../../atoms/Legend/Legend';
+import Text from '../../atoms/Text/Text';
+
+const CheckboxWrapper = styled.div`
+  padding: 4px 0;
+`;
+
+interface ICheckboxGroupFieldItem {
+  label: string;
+  name: string;
+  defaultChecked?: boolean;
+  disabled?: boolean;
+}
+
+export interface ICheckboxGroupFieldProps {
+  label: string;
+  items: ICheckboxGroupFieldItem[];
+  onChange: (value: Record<string, boolean>) => void;
+  disabled?: boolean;
+  value?: Record<string, boolean>;
+}
+
+const CheckboxGroupField = ({ items, label, onChange, value, disabled }: ICheckboxGroupFieldProps) => {
+  const [innerValue, setInnerValue] = useState<Record<string, boolean>>(
```

Comment:
This could be a custom hook, also I am not sure about having a reduce in the initial state

Reply:
how else would you initialize though?

Reply:
as far as the hook goes, I'm not sure if abstracting it out would bring in any value as this `handleChange` is pretty specific to `CheckboxGroupField` and can't be reused anywhere else

Reply:
I think we can address this refactor later if needed as new PR 👍🏻

Reply:
You make a custom hook that takes in items as an argument, declare a const that takes in the result of the reduce, then initialise the useState with that value (also might be easier to test that way). Then return the state and setState at the end of the hook.

Reply:
That's true but I'm not sure if extracting the hook makes that much of a difference. I personally prefer to keep everything in the component unless either the component is getting unwieldy or things can be reused. However, this is just a personal preference and if you feel it improves the code, go for it